flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/conversion/HiveInspectors.java

```diff
@@ -450,4 +453,45 @@ private static ObjectInspector getObjectInspector(TypeInfo type) {
 	public static DataType toFlinkType(ObjectInspector inspector) {
 		return HiveTypeUtil.toFlinkType(TypeInfoUtils.getTypeInfoFromTypeString(inspector.getTypeName()));
 	}
+
+	// given a Hive ObjectInspector, get the class for corresponding Flink object
+	private static Class<?> classForObjectInspector(ObjectInspector inspector) {
+		switch (inspector.getCategory()) {
+			case PRIMITIVE: {
+				PrimitiveObjectInspector primitiveOI = (PrimitiveObjectInspector) inspector;
+				switch (primitiveOI.getPrimitiveCategory()) {
```

Comment:
quite a few data types are missing, e.g. short, date, timestamp, binary, decimal, etc. But we may end up not needing this if the above suggested approach works